java/test/jmri/jmrix/loconet/LnOpsModeProgrammerTest.java

```diff
@@ -466,6 +466,78 @@ public void testOpsReadLocoNetMode() throws ProgrammerException {
+        lnopsmodeprogrammer.readCV("12", pl);
+
+        // should have written
+        Assert.assertEquals("one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("No programming reply", 0, pl.getRcvdInvoked());
+
+        Assert.assertEquals("sent", "EF 0E 7C 2F 00 00 04 00 00 0B 00 7F 7F 00", lnis.outbound.get(0).toString());
+
+        // check echo of sent message has no effect
+        LocoNetMessage m = lnis.outbound.get(0);
+        lnopsmodeprogrammer.message(m);
+        Assert.assertEquals("still one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("No programming reply", 0, pl.getRcvdInvoked());
+
+        // LACK "command rejected" followed by Known-good message in reply
+        m = new LocoNetMessage(new int[]{0xB4, 0x6F, 0x00, 0x24});
+        lnopsmodeprogrammer.message(m);
+        sm.message(m);
+
+        m = new LocoNetMessage(new int[]{0xE7, 0x0E, 0x7C, 0x2B, 0x00, 0x00, 0x02, 0x47, 0x00, 0x1C, 0x23, 0x7F, 0x7F, 0x3B});
+        lnopsmodeprogrammer.message(m);
+        sm.message(m);
+        JUnitUtil.waitFor(()->{return pl.getRcvdInvoked() == 1;},"getRcvdInvoked not set");
+        Assert.assertEquals("still one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("Reply status OK", 0, pl.getRcvdStatus());
+
+     }
+
+     @Test
+     public void testOpsReadLocoNetModeLACKNotImplemented() throws ProgrammerException {
+        // allow transponding
+        sm.setTranspondingAvailable(false);
+
+        LnOpsModeProgrammer lnopsmodeprogrammer = new LnOpsModeProgrammer(memo, 4, true);
+
+        lnopsmodeprogrammer.setMode(LnProgrammerManager.LOCONETOPSBOARD);
+        lnopsmodeprogrammer.readCV("12", pl);
+
+        // should have written
+        Assert.assertEquals("one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("No programming reply", 0, pl.getRcvdInvoked());
+
+        Assert.assertEquals("sent", "EF 0E 7C 2F 00 00 04 00 00 0B 00 7F 7F 00", lnis.outbound.get(0).toString());
+
+        // check echo of sent message has no effect
+        LocoNetMessage m = lnis.outbound.get(0);
+        lnopsmodeprogrammer.message(m);
+        Assert.assertEquals("still one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("No programming reply", 0, pl.getRcvdInvoked());
+
+        // LACK "function not implemented" followed by Known-good message in reply
+        m = new LocoNetMessage(new int[]{0xB4, 0x6F, 0x7F, 0x5B});
+        lnopsmodeprogrammer.message(m);
+        sm.message(m);
+
+        m = new LocoNetMessage(new int[]{0xE7, 0x0E, 0x7C, 0x2B, 0x00, 0x00, 0x02, 0x47, 0x00, 0x1C, 0x23, 0x7F, 0x7F, 0x3B});
+        lnopsmodeprogrammer.message(m);
+        sm.message(m);
+        JUnitUtil.waitFor(()->{return pl.getRcvdInvoked() == 1;},"getRcvdInvoked not set");
+        Assert.assertEquals("still one message sent", 1, lnis.outbound.size());
+        Assert.assertEquals("Reply status OK", 0, pl.getRcvdStatus());
+
+     }
+
```

Comment:
Need additional tests for writeCV and confirmCV.

Reply:
@devel-bobm there were no existing tests for LoconetOpsMode writeCV or confirmCV so I didn't have a pattern to follow.  I am very hesitatant to create new ones since I don't really understand them and don't know how to check them.

Reply:
This needs to get resolved.  I cannot guess when I can work on this.  Perhaps today.  Perhaps later this week.  Difficult to guess right now!

Reply:
@devel-bobm Thanks.  I thought I could figure it out between the existing testOpsReadLoconNetMode() and testOpsWriteDecoder(). but the latter test does not exist.  There aren't any Confirm tests, perhaps because this is just a Read as far as Loconet is concerned?